tests/atlasgen/test_stacks.py

```diff
@@ -0,0 +1,49 @@
+import os
+import tempfile
+from pathlib import Path
+
+import numpy as np
+import pytest
+import tifffile
+
+from brainglobe_atlasapi.atlas_generation.stacks import write_stack
+
+
+@pytest.fixture
+def image_unit():
+    """Fixture providing a dummy image stack."""
+    image_unit = np.random.randint(0, 65535, (10, 10, 10), dtype=np.uint16)
+    yield image_unit
+
+
+@pytest.fixture
+def image_float():
+    """Fixture providing a dummy image stack."""
+    image_float = np.random.rand(10, 10, 10).astype(np.float32)
+    yield image_float
+
+
+@pytest.fixture
+def tmp_path():
+    """Fixture providing a temporary directory for testing."""
+    with tempfile.TemporaryDirectory() as tmpdir:
+        yield tmpdir
+
+
+def test_write_stack(image_unit, image_float, tmp_path):
```

Comment:
I suggest splitting this into two tests, one for the integer image and one for the floating point image.